File: src/sort/selectionSort.go
package sort

// SelectionSort
func SelectionSort(nums []int) []int {
    n := len(nums)
    for i := 0; i < n-1; i++ {
        minIndex := i
        for j := i + 1; j < n; j++ {
            if nums[j] < nums[minIndex] {
                minIndex = j
            }
        }

        if i != minIndex {
            nums[i], nums[minIndex] = nums[minIndex], nums[i]
        }
    }

    return nums
}

// Please write the main function below.

// import (
// 	"algorithm/src/sort"
// 	"fmt"
// )

// func main() {
// arr := []int{12, 4, 7, 2, 10, 8}
// sort.SelectionSort(arr)
// var N int
// fmt.Print("Enter a number: ")
// fmt.Scan(&N)
// result := recursive.Factorial(N)
// fmt.Println(result)
// }

File: src/sort/selectionSort.ts
// SelectionSort
function selectionSort(arr: number[]): void {
	const N: number = arr.length;
	for (let i = 0; i < N - 1; i++) {
		let minPosition: number = i;
		let minVal: number = arr[i];
		for (let j = i + 1; j < N; j++) {
			if (arr[j] < minVal) {
				minPosition = j;
				minVal = arr[j];
			}
		}
		[arr[i], arr[minPosition]] = [arr[minPosition], arr[i]];
	}
}
